src/wazuh_modules/wm_exec.c

```diff
@@ -222,57 +237,59 @@ DWORD WINAPI Reader(LPVOID args) {
 // Add process to pool
 
 void wm_append_handle(HANDLE hProcess) {
-    int i;
+    HANDLE * p_hProcess = NULL;
 
-    w_mutex_lock(&wm_children_mutex);
+    os_calloc(1, sizeof(HANDLE), p_hProcess);
+    *p_hProcess = hProcess;
 
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (!wm_children[i]) {
-            wm_children[i] = hProcess;
-            break;
+    if (wm_children_list) {
+        if (OSList_AddData(wm_children_list, (void *)p_hProcess)) {
+            return;
```

Comment:
Possible missing lock.

```suggestion
        w_mutex_lock(&wm_children_mutex);
        void * retval = OSList_AddData(wm_children_list, (void *)p_hProcess);

        w_mutex_unlock(&wm_children_mutex);
        if (retval != NULL) {
            return;
```